src/index.tsx

```diff
@@ -4,49 +4,109 @@ import * as ReactDOM from "react-dom";
 import { createStore, applyMiddleware } from "redux";
 import * as App from "grommet/components/App";
 import ConfigForm from "./components/ConfigForm";
-import { SpyReducer } from "./reducers/SpyReducer";
+import { SpyReducer, XCSpyState } from "./reducers/SpyReducer";
 import * as logger from "redux-logger";
 import thunk from "redux-thunk";
-import CompositionModel from "./components/CompositionModel";
+import Components from "./components/Components";
 import { setCompositionModel } from "actions/compositionModel";
 import sessionXCSpy from "utils/sessionXCSpy";
+import { Dispatch } from "redux";
+import { Parser } from "utils/parser";
+import { subscribeAllStateMachines, snapshotEntryPoint } from "core";
+import SideBar from "components/SideBar";
+import * as Split from "grommet/components/Split";
+import * as Box from "grommet/components/Box";
+import AppHeader from "components/AppHeader";
+import Footer from "components/Footer";
+import TransitionProperties from "components/TransitionProperties";
+import StateMachineProperties from "components/StateMachineProperties";
 
 const middleware = applyMiddleware(thunk, logger());
 const store = createStore(SpyReducer, middleware);
 
-let App = (props) => {
-  if (!props.submitted) {
-    return (
-      <ConfigForm />
-    );
+interface XCSpyGlobalProps extends XCSpyProps, XCSpyCallbackProps {
+};
+
+interface XCSpyProps {
+  submitted: boolean;
+  selectedApi: string;
+  serverUrl: string;
+  compositionModel: any;
+};
+
+interface XCSpyCallbackProps {
+  setCompositionModel: (xcApiName: string, serverUrl: string) => void;
+};
+
+class XCSpyApp extends React.Component<XCSpyGlobalProps, XCSpyState> {
```

Comment:
You don't need to define a class. Prefer lambda expression definition
const XCSpyApp = (props) => {...